### PR TITLE
gomemif: handle EINTR error

### PR DESCRIPTION
**Pull request by Giluerre**
-Add handling for EINT error
-Fix packet burst
-Fix connection between interfaces in [(intr/intr),(intr,poll),(poll,intr)] modes 
-Fix master interface

Signed-off-by: Daniel Béreš <dberes@cisco.com>

### CODE REVIEW

File: extras/gomemif/memif/interface.go [hunk 1]
```diff
@@ -367,6 +368,22 @@ func eventFd() (efd int, err error) {
 	return int(u_efd), nil
 }
 
+// The commented function is advance preparation for potential replacement of EpollWait syscall
+// func EpollPWait(epfd int, events []EpollEvent, msec int, sigmask sigmask) (n int, err error) {
```
Comment (sknat) [inline extras/gomemif/memif/interface.go:372]: We can keep this function in preparation, but we should probably not comment it out

File: extras/gomemif/memif/packet_reader.go [hunk 2]
```diff
@@ -89,3 +97,92 @@ refill:
...
+		return 0, nil
+	}
+
+	rx := 0
+	for nSlots > 0 {
+		// copy descriptor from shm
+		q.getDescBuf(slot&mask, desc)
+		length = desc.getLength()
+		offset = desc.getOffset()
+		//b0 := &pkt[rx]
```
Comment (sknat) [inline extras/gomemif/memif/packet_reader.go:133]: commented code

File: extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go [hunk 3]
```diff
@@ -68,123 +73,130 @@ func Responder(i *memif.Interface) error {
...
 	// allocate packet buffer
-	pkt := make([]byte, 2048)
 	// get rx queue
 	rxq0, err := i.GetRxQueue(0)
 	_ = err
-
-	// read packet from shared memory
-	pktLen, err := rxq0.ReadPacket(pkt)
-	_, _ = err, pktLen
+	//TODO: fix Refill() to replace ReadPacket()
```
Comment (sknat) [inline extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go:196]: We should probably address this before merging this

File: extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go [hunk 4]
```diff
@@ -68,123 +73,130 @@ func Responder(i *memif.Interface) error {
...
+			}
+			icmpLayer := gopkt.Layer(layers.LayerTypeICMPv4)
+			if icmpLayer == nil {
+				fmt.Println("Missing ICMPv4 layer.")
+			}
+			icmp, _ := icmpLayer.(*layers.ICMPv4)
+			if icmp.TypeCode.Type() != layers.ICMPv4TypeEchoRequest {
+				fmt.Println("Not ICMPv4 echo request.")
+			}
+			//fmt.Println("Received an ICMPv4 echo request.")
```
[the remark this answers is not in the document]
Comment (Giluerre) [reply extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go]: Fixed

File: extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go [hunk 5]
```diff
@@ -68,123 +73,130 @@ func Responder(i *memif.Interface) error {
...
-				if icmpLayer == nil {
-					fmt.Println("Missing ICMPv4 layer.")
-				}
-				icmp, _ := icmpLayer.(*layers.ICMPv4)
-				if icmp.TypeCode.Type() != layers.ICMPv4TypeEchoRequest {
-					fmt.Println("Not ICMPv4 echo request.")
-				}
-				fmt.Println("Received an ICMPv4 echo request.")
+	for i := 0; i < int(nPackets); i++ {
+		//fmt.Printf("pktLen: %d\n", pkt[i].Buflen)
```
[the remark this answers is not in the document]
Comment (Giluerre) [reply extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go]: Fixed

File: extras/gomemif/memif/interface.go (hunk 1 above)
Comment (Giluerre) [reply extras/gomemif/memif/interface.go]: Fixed

File: extras/gomemif/memif/packet_reader.go (hunk 2 above)
Comment (Giluerre) [reply extras/gomemif/memif/packet_reader.go]: Fixed

File: extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go (hunk 3 above)
Comment (Giluerre) [reply extras/gomemif/examples/icmp_responder_cb/icmp_responder_cb.go]: Fixed: I just forgot to delete the comment. Refill() already replaced ReadPacket() in refilling queues.

**Review by sknat — APPROVED** [review]

**Pull request merged**